src/rust/scheduler/scheduler.rs

```diff
@@ -402,39 +421,82 @@ mod tests {
+        let task_id: u64 = handle.get_task_id();
+        match scheduler.from_task_id(task_id) {
+            Some(retreived_task_handle) => crate::ensure_eq!(task_id, retreived_task_handle.get_task_id()),
+            None => anyhow::bail!("from_task_id() must not return None"),
+        }
+        Ok(())
+    }
 
-        // Ensure that the first task has completed.
-        crate::ensure_eq!(handle.has_completed(), true);
+    #[test]
+    fn remove_removes_task_id_as_well() -> Result<()> {
+        let scheduler: Scheduler = Scheduler::default();
+        const NUM_TASKS: usize = 8192;
+        let mut handles: Vec<TaskHandle> = Vec::<TaskHandle>::with_capacity(NUM_TASKS);
 
-        // Create another task.
-        let task2: DummyTask = DummyTask::new(String::from("testing"), Box::pin(DummyCoroutine::new(0)));
-        let handle2: TaskHandle = match scheduler.insert(task2) {
-            Some(handle) => handle,
-            None => anyhow::bail!("insert() failed"),
-        };
-        let task_id2: u64 = handle2.get_task_id();
+        crate::ensure_eq!(true, scheduler.task_ids.borrow().is_empty());
 
-        // Ensure that the second task has a unique id.
-        crate::ensure_neq!(task_id2, task_id);
+        for val in 0..NUM_TASKS {
+            let task: DummyTask = DummyTask::new(String::from("testing"), Box::pin(DummyCoroutine::new(val)));
+            let handle: TaskHandle = match scheduler.insert(task) {
+                Some(handle) => handle,
+                None => panic!("insert() failed"),
+            };
+            handles.push(handle);
+        }
+
+        // This poll is required to give the opportunity for all the tasks to complete.
+        scheduler.poll();
+
+        // Remove tasks one by one and check if remove is only removing the task requested to be removed.
+        let mut curr_num_tasks: usize = NUM_TASKS;
+        for i in 0..NUM_TASKS {
+            let task_id: u64 = handles[i].get_task_id();
+            crate::ensure_eq!(true, scheduler.task_ids.borrow().contains_key(&task_id));
+            scheduler.remove(&handles[i]);
+            curr_num_tasks = curr_num_tasks - 1;
+            crate::ensure_eq!(scheduler.task_ids.borrow().len(), curr_num_tasks);
+            crate::ensure_eq!(false, scheduler.task_ids.borrow().contains_key(&task_id));
+        }
+
+        crate::ensure_eq!(scheduler.task_ids.borrow().is_empty(), true);
 
         Ok(())
     }
 
     #[bench]
-    fn bench_scheduler_poll(b: &mut Bencher) {
+    fn bench_insert(b: &mut Bencher) {
+        let scheduler: Scheduler = Scheduler::default();
+
+        b.iter(|| {
+            let task: DummyTask =
+                DummyTask::new(String::from("testing"), Box::pin(black_box(DummyCoroutine::default())));
+            let handle: TaskHandle = scheduler.insert(task).expect("couldn't insert future in scheduler");
+            black_box(handle);
+        });
+    }
+
+    #[bench]
+    fn bench_poll(b: &mut Bencher) {
```

Comment:
Could you add a Rust Doc comment for this test, like we have for other tests in this module?